src/panels/ai/ChatPanelSection.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState } from 'react';
+import { VscMegaphone, VscRunAll, VscWand } from 'react-icons/vsc';
+
+import { chatWithAI } from '../../api/chat';
+import { PanelSection } from '../../components/PanelSection';
+import { cx } from '../../cx';
+
+interface ChatThreadItem {
+  id: string;
+  question: string;
+  answer?: string;
+}
+type ChatThread = ChatThreadItem[];
+
+export function ChatPanelSection() {
+  const [text, setText] = useState('');
+
+  const [chatThread, setChatThread] = useState<ChatThread>([
+    // {
+    //   id: '1',
+    //   question: 'What is your name?',
+    //   answer:
+    //     'I am an AI language model and do not have a name. How may I assist you with your query related to the given context?',
+    // },
+    // {
+    //   id: '2',
+    //   question: 'What can you tell me about hidden feedback loops in machine learning?',
+    //   answer:
+    //     'Hidden feedback loops in machine learning refer to situations where two systems indirectly influence each other through the world, leading to changes in behavior that may not be immediately visible. These loops may exist between completely disjoint systems and can make analyzing the effect of proposed changes extremely difficult, adding cost to even simple improvements. It is recommended to look carefully for hidden feedback loops and remove them whenever feasible.',
+    // },
+  ]);
+  const [currentChatThreadItem, setCurrentChatThreadItem] = useState<ChatThreadItem | null>(null);
+
+  function handleKeyDown(evt: React.KeyboardEvent<HTMLTextAreaElement>): void {
+    if (!evt.shiftKey && evt.code === 'Enter') {
+      evt.preventDefault();
+      handleChat(text);
+      return;
+    }
+  }
+
+  function handleChat(question: string) {
+    if (!question || currentChatThreadItem) {
+      return;
+    }
+    setCurrentChatThreadItem({ id: String(chatThread.length + 1), question: text });
+    setText('');
+    chatWithAI(question)
+      .then((reply) => {
+        setChatThread([...chatThread, { id: String(chatThread.length + 1), question: text, answer: reply[0] }]);
```

Comment:
There may be a stale closure issue here if you send two chats before the first one comes back. Safer to use the function form of `setState`:

```ts
setChatThread(prevChatThread => [...prevChatThread, { id: String(prevChatThread.length + 1), question: text, answer: reply[0] }]);
```

(also below)